### DISCUSSION TITLE
AK4556VT with Yummy dsp

### DISCUSSION THREAD

**Discussion by GazTurtle** [garygru/yummyDSP#12] [Q&A]
Hi Gary, last year I used your fabulous library with an 'adafruit huzzah 32', I had it almost using I2s, DMA and the built in DAC and ADC's.

I ended up with a terrible jitter and couldn't find a solution. Trawled the internet and found no-one else had managed to solve the timing issues I faced.

I gave up and used a really simple solution still using built in ADC's and DAC's but sampling the ADC outside of the I2S connection to the ADC. It was fun and let me learn the rest of your library.

Now I am going to follow you more closely and use the AK4556VT as you did. I have one soldered up and ready to prototype. 

I wonder whether you could point me in the direction of wiring up the chip? I have looked at your Yummy Cookie for the AK4556VT, which I wish I could buy :) 

I am being lazy and wanting to avoid reading all the data sheet for the AK4556VT chip and stand on the shoulders of your work with it so far. 

Many thanks for teaching me how to code DSP, really enjoyed playing with your library. The little synth I built now sits proudly in my eurorack modular synth housing and makes a lovely Karplus–Strong string synthesis 'boiiiinnnnnggggg'.

**Comment by garygru** [comment]
Hi,
great to hear that the framework found its way into your eurorack module. Sorry can't help you right away. I'll have to look that up myself. You can probably see all the details in the kicad schematics on [hackaday](https://hackaday.io/garyfon) and there are also some comments in the audio driver code.  A quick look at the schematic in data sheet will also help. But yeah I admit, I should upload the schematics as PDF. Give me some time. When you have a question to a specific pin, I probably can help. cheers, gary

**Reply by GazTurtle** [reply]
thanks for the reply Gary. I shall follow your links and crack on with it. I really ought to read the datasheet anyway. When you get chance, did you create the Yummy Cookie for the AK4556VT from the example application circuit in the datasheet?

**Reply by GazTurtle** [reply]
Well it couldn't have been easier! I used a UDA1334 breakout board from adafruit...

https://www.adafruit.com/product/3678 

And connected it this way...

// Audio Amp Pins
#define I2S_BCK_PIN         26 //connect to BCLK on UDA1334
#define I2S_LRCLK_PIN     27 //connect to WSEL on UDA1334

#define I2S_DOUT_PIN      14 // I connect this to the DIN on the UDA1334
#define I2S_DIN_PIN          13 //not connected for UDA1334

#define CODEC_ENABLE_PIN 33 //not connected for my tests

and it worked wonderfully with the Synth example in your library.

Thank you so much Gary.
